routes/redis/SetState.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * Created by Kun on 6/28/2015.
+ */
+
+//http://www.sitepoint.com/using-redis-node-js/
+
+
+var express = require("express");
+var Q = require('q');
+var router = express.Router();
+var redis = require('redis');
+var config = require("konfig")();
+
+var port=config.app.redis.port;
+var host=config.app.redis.host;
+
+var client = redis.createClient(port,host);
+
+
+router.route('/:cookie/:username/:LastPage').get(function(req, res) {
```

Comment:
We will want to be able to take in and store a state object, not a LastPage. We should also be able to get the username from the cookie at any given time since it is always being stored.

I also recommend taking the hmset stuff and moving it to a service in impservices so that both your route and Trevor's login route can store the stuff in the same way. His login will save the cookie, the username, and an empty state object initially.